diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt

```diff
@@ -107,6 +116,24 @@ open class DiktatJavaExecTaskBase @Inject constructor(
     @Suppress("FUNCTION_BOOLEAN_PREFIX")
     override fun getIgnoreFailures(): Boolean = ignoreFailuresProp.getOrElse(false)
 
+    private fun createReporterFlag(diktatExtension: DiktatExtension): String {
+        val flag: StringBuilder = StringBuilder()
+
+        // Plain, checkstyle and json reporter are provided out of the box in ktlint
+        when(diktatExtension.reporterType) {
+            "json" -> flag.append("--reporter=json")
+            "html" -> flag.append("--reporter=html")
+            "checkstyle" -> flag.append("--reporter=checkstyle")
+            else -> flag.append("--reporter=plain")
```

Comment:
I think they already have `baseline` reporter (or was it added in 0.40.0?) Still, let's log a warning if unknown reporter is specified and then fallback to plain.

Reply:
Also, one could imagine a use case, when end user adds a custom reporter as a dependency in our plugin's configuration and then specifies it's name (need to see, how it's loaded in ktlint). So it might be more versatile to allow arbitrary names here and then let ktlint validate them.

Reply:
We need to specify it here if a user will provide us an output here